eng/scripts/InstallVisualStudio.ps1

```diff
@@ -103,10 +102,60 @@ Write-Host "Installing Visual Studio 2019 $Edition" -f Magenta
 Write-Host ""
 Write-Host "Running '$bootstrapper $arguments'"
 
-$process = Start-Process -FilePath "$bootstrapper" -ArgumentList $arguments `
-    -PassThru -RedirectStandardError "$intermedateDir\errors.txt" -Verbose -Wait
-if ($process.ExitCode -ne 0) {
-    Get-Content "$intermedateDir\errors.txt" | Write-Error
+foreach ($i in 0, 1, 2) {
+    $process = Start-Process -FilePath "$bootstrapper" -ArgumentList $arguments -ErrorAction Continue -PassThru `
+        -RedirectStandardError "$intermedateDir\errors.txt" -Verbose -Wait
+    Write-Host "Exit code = $($process.ExitCode)."
+    if ($process.ExitCode -eq 0) {
+        break
+    } else {
+        # https://docs.microsoft.com/en-us/visualstudio/install/use-command-line-parameters-to-install-visual-studio#error-codes
+        if ($process.ExitCode -eq 3010) {
+            Write-Error "Installation requires restart to finish the VS update."
```

Comment:
`Write-Error` is fatal (equivalent to 'throw Exception') when  `$ErrorActionPreference = 'Stop'`. Use `write-warning` or `write-host -f Red "..."` instead

Reply:
Changed the preference [above](https://github.com/aspnet/AspNetCore/pull/9300/files#diff-27fbe4ead004bbb25de7d0b254c584fdR105) instead

Reply:
Changing $ErrorActionPreference in the middle of a script seems to me like a bad pattern. Most good powershell code I've read leaves `$ErrorActionPreference = 'Stop'` throughout the whole script, only using `-ErrorAction continue` when appropriate. But I've also never seen `Write-Error -ErrorAction continue` because you can just do `Write-warning`.

Reply:
:ok: I'll add the requisite `-ErrorAction continue` parameters and switch to `Write-Warning`